statement_test.go

```diff
@@ -73,9 +184,104 @@ func TestPrepareQueryNamed(t *testing.T) {
 	require.Equal(t, 1, foo2)
 
 	require.NoError(t, prepared.Close())
+
+	// Prepare on a connection.
+	c, err := db.Conn(context.Background())
+	require.NoError(t, err)
+
+	prepared, err = c.PrepareContext(context.Background(), `SELECT * FROM foo WHERE bar = $bar AND baz = $baz`)
+	require.NoError(t, err)
+	res, err := prepared.Query(sql.Named("bar", "hello"), sql.Named("baz", 0))
+	require.NoError(t, err)
+	require.NoError(t, res.Close())
+	require.NoError(t, prepared.Close())
+
+	// Access the raw connection & statement.
+	err = c.Raw(func(driverConn interface{}) error {
+		conn := driverConn.(*Conn)
+		s, err := conn.PrepareContext(context.Background(), `INSERT INTO foo VALUES ($bar, $baz)`)
+		require.NoError(t, err)
+		stmt := s.(*Stmt)
+
+		stmtType, err := stmt.StatementType()
+		require.NoError(t, err)
+		require.Equal(t, DUCKDB_STATEMENT_TYPE_INSERT, stmtType)
+
+		paramName, err := stmt.ParamName(0)
+		require.ErrorContains(t, err, paramIndexErrMsg)
+		require.Equal(t, "", paramName)
+
+		paramName, err = stmt.ParamName(1)
+		require.NoError(t, err)
+		require.Equal(t, "bar", paramName)
+
+		paramName, err = stmt.ParamName(2)
+		require.NoError(t, err)
+		require.Equal(t, "baz", paramName)
+
+		paramName, err = stmt.ParamName(3)
+		require.ErrorContains(t, err, paramIndexErrMsg)
+		require.Equal(t, "", paramName)
+
+		paramType, err := stmt.ParamType(0)
+		require.ErrorContains(t, err, paramIndexErrMsg)
+		require.Equal(t, TYPE_INVALID, paramType)
+
+		paramType, err = stmt.ParamType(1)
+		// Will be fixed in the next release.
+		// https://github.com/duckdb/duckdb/pull/14952
+		// require.Equal(t, TYPE_VARCHAR, paramType)
+		require.Equal(t, TYPE_INVALID, paramType)
+		require.NoError(t, err)
+
+		paramType, err = stmt.ParamType(2)
+		// Will be fixed in the next release.
+		// https://github.com/duckdb/duckdb/pull/14952
+		// require.Equal(t, TYPE_INTEGER, paramType)
+		require.Equal(t, TYPE_INVALID, paramType)
+		require.NoError(t, err)
```

Comment:
Let's put a `FIXME` here with a link to an issue (can you open one?). So we don't forget to come back to this.

Reply:
`require.Equal(t, TYPE_INVALID, paramType)` will fail when upgrading to the new DuckDB version, ensuring that we won't overlook this.